Makefile

```
@@ -4,6 +4,8 @@ DEFAULT_IMAGE_VERSION ?= $(BASE_BRANCH)
 export BASE_BRANCH
 export DEFAULT_IMAGE_VERSION
 
+DEFAULT_REPO = "quay.io/submariner"
```

```suggestion
DEFAULT_REPO ?= "quay.io/submariner"
```

Won't this be better, to allow builder to dynamically specify the repo during build time?

---

`make DEFAULT_REPO=…` overrides this as-is; `?=` is useful to keep an existing variable value that might have been set in a Makefile (_e.g._ through an `include`).

---

AFAIK `?=` supports values from the env while `=` would overwrite them, so I still think it'll be more useful.

e.g. `DEFAULT_REPO=… make ...` won't work unless it's assigned via `?=`

---

Ah yes, that’s a good point, so `?=` would indeed be better.

---

well, `make DEFAULT_REPO="quay.io/submariner2" cmd/bin/subctl` works fine 
`Building 'cmd/bin/subctl-devel-db0dc08-linux-amd64' (ldflags: $'-X github.com/submariner-io/subctl/pkg/version.Version=devel-db0dc08 \t       -X github.com/submariner-io/submariner-operator/api/submariner/v1alpha1.DefaultSubmarinerOperatorVersion=devel \t       -X github.com/submariner-io/submariner-operator/api/submariner/v1alpha1.DefaultRepo=quay.io/submariner2')`

without the `?=` but I have anyways assigned it like that.

---

> without the `?=` but I have anyways assigned it like that.

Where? The current PR doesnt reflect that

---

Now it does.